code/game/machinery/shieldgen.dm

```diff
@@ -247,6 +247,19 @@
 	var/obj/structure/cable/attached		// the attached cable
 	var/storedpower = 0
 
+/obj/machinery/shieldwallgen/science		//Use for xenobiology on maps where you have shield wall generators in xeno.
+	name = "shield wall generator"
```

Comment:
I thought about that when I was about to go to bed- I'll rename it to xenobiology shield wall generator when I get home.